### PR TITLE
[PREVIEW] SIDM-1840: Get URL extraction working

### PR DESCRIPTION
**Pull request by steveclewer**
https://tools.hmcts.net/jira/browse/SIDM-1840

Use URL extracted from email for password reset.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/js/shared/custom_steps.js [hunk 1]
```diff
@@ -3,23 +3,57 @@ var TestData = require('../config/test_data');
 module.exports = function() {
   return actor({
 
-    lockAccount: function (email) {
+    lockAccount: function (email, serviceName) {
+        var registerPage = TestData.WEB_PUBLIC_URL + '/users/selfRegister?redirect_uri=https://www.autotest.com&client_id=' + serviceName;
+        var loginPage = TestData.WEB_PUBLIC_URL + '/login?redirect_uri=https://www.autotest.com&client_id=' + serviceName + 'state=';
```
Comment (jburke-amido) [inline src/test/js/shared/custom_steps.js]: needs to be "&state=" ?

Comment (steveclewer) [reply src/test/js/shared/custom_steps.js]: Not sure how that got removed - added back

File: src/test/js/account_lockout_reset_password_test.js [hunk 2]
```diff
@@ -38,13 +38,29 @@ return Promise.all([
...
-       I.see('Your account is locked due to too many unsuccessful attempts.');
+       I.lockAccount(citizenEmail, serviceName);
        I.click('reset your password');
        I.waitForText('Reset your password', 20, 'h1');
        I.fillField('#email', citizenEmail);
        I.click('Submit');
        I.waitForText('Check your email', 20, 'h1');
-       await I.verifyEmailSent(citizenEmail);
+       I.see('Check your email');
+       I.wait(20);
```
Comment (jburke-amido) [inline src/test/js/account_lockout_reset_password_test.js]: for testing?

Comment (steveclewer) [reply src/test/js/account_lockout_reset_password_test.js]: Yeah it's from testing. Will remove

File: src/test/js/shared/idam_helper.js [hunk 3]
```diff
@@ -241,6 +241,33 @@ class IdamHelper extends Helper {
...
+  extractUrl(email) {
+     return (notifyClient
+         .getNotifications("email", "sending")
+         .then(response => {
+             console.log("Searching " + response.body.notifications.length + " emails(s)");
+             var result = response.body.notifications.find(obj => {
+                 if (obj.email_address === email) {
+                     return obj.email_address === email
+                 } else {
+                     console.log("ignoring unmatched email address " + obj.email_address);
```
Comment (jburke-amido) [inline src/test/js/shared/idam_helper.js]: best comment this out - could potentially put real email addresses in the log

Comment (steveclewer) [reply src/test/js/shared/idam_helper.js]: Yep, was debug

Comment (jburke-amido) [reply src/test/js/shared/idam_helper.js]: Actually, can you remove the whole else block and put in a comment that says "NOTE:  NEVER LOG EMAIL ADDRESSES FROM THE PRODUCTION QUEUE". Once we create the other notify keys it won't be an issue, but for now...

File: src/test/js/shared/custom_steps.js [hunk 4]
```diff
@@ -3,23 +3,57 @@ var TestData = require('../config/test_data');
...
+        this.see('Sign in');
         this.fillField('username', email);
         this.fillField('password', '111');
         this.click('Sign in');
-        this.wait(1);
+        this.waitForText('Incorrect email or password', 20, 'h2');
+        this.clearCookie();
+        // Second
+        this.amOnPage(loginPage);
+        this.dontSee('Incorrect email or password');
```
Comment (jburke-amido) [inline src/test/js/shared/custom_steps.js]: think we need waitInUrl(...) here before the dontSee

File: src/test/js/shared/idam_helper.js [hunk 5]
```diff
@@ -9,10 +9,10 @@ if (process.env.PROXY_SERVER) {
   agentToUse = new HttpsProxyAgent(process.env.PROXY_SERVER);
 } else {
   console.log('using real agent');
-  const Https = require('https');
-  agentToUse = new Https.Agent({
-    rejectUnauthorized: false
-  });
+//  const Https = require('https');
```
Comment (jburke-amido) [inline src/test/js/shared/idam_helper.js]: Can you drag in all the improved code about agent and keys from admin's helper

File: src/test/js/shared/idam_helper.js [hunk 6]
```diff
@@ -241,6 +241,33 @@ class IdamHelper extends Helper {
...
+  extractUrl(email) {
+     return (notifyClient
+         .getNotifications("email", "sending")
+         .then(response => {
+             console.log("Searching " + response.body.notifications.length + " emails(s)");
+             var result = response.body.notifications.find(obj => {
+                 if (obj.email_address === email) {
+                     return obj.email_address === email
+                 } else {
+                     console.log("Ignoring unmatched email address");
```
Comment (jburke-amido) [inline src/test/js/shared/idam_helper.js]: Remove the whole else block and replace with a comment that says "NOTE: NEVER LOG EMAIL ADDRESS FROM THE PRODUCTION QUEUE".
Won't be an issue once we create a new notify api key for testing

File: src/test/js/account_lockout_reset_password_test.js [hunk 7]
```diff
@@ -34,17 +34,32 @@ return Promise.all([
...
+       I.waitForText('Your password has been changed', 20, 'h1');
+       I.see('You can now sign in with your new password.')
+       I.amOnPage(TestData.WEB_PUBLIC_URL + '/users/selfRegister?redirect_uri=https://idam.testservice.gov.uk&client_id=' + serviceName);
+       I.click('Sign in to your account');
+       I.waitInUrl('/login', 180);
+       I.waitForText('Sign in or create an account', 20, 'h1');
+       I.fillField('#username', citizenEmail);
+       I.fillField('#password', 'Passw0rd1234');
+       I.click('Sign in');
+       I.waitInUrl('idam.testservice.gov.uk', 180);
```
Comment (jburke-amido) [inline src/test/js/account_lockout_reset_password_test.js]: check for "code" parameter, and "error" not being present

**Pull request merged**